src/gobstuf/mks_utils.py

```diff
@@ -47,37 +127,106 @@ def _dd(cls, mks_datum):
             return f"{cls._yyyy(mks_datum)}-{cls._mm(mks_datum)}-{cls._dd(mks_datum)}"
 
     @classmethod
-    def as_jaar(cls, mks_datum):
-        if cls._is_mks_datum(mks_datum):
+    def as_jaar(cls, mks_datum, ind_onvolledige_datum=None):
+        if cls._is_mks_datum(mks_datum) and IncompleteDateIndicator(ind_onvolledige_datum).is_jaar_bekend():
             return int(cls._yyyy(mks_datum))
 
     @classmethod
-    def as_maand(cls, mks_datum):
-        if cls._is_mks_datum(mks_datum):
+    def as_maand(cls, mks_datum, ind_onvolledige_datum=None):
+        if cls._is_mks_datum(mks_datum) and IncompleteDateIndicator(ind_onvolledige_datum).is_maand_bekend():
             return int(cls._mm(mks_datum))
 
     @classmethod
-    def as_dag(cls, mks_datum):
-        if cls._is_mks_datum(mks_datum):
+    def as_dag(cls, mks_datum, ind_onvolledige_datum=None):
+        if cls._is_mks_datum(mks_datum) and IncompleteDateIndicator(ind_onvolledige_datum).is_dag_bekend():
             return int(cls._dd(mks_datum))
 
     @classmethod
-    def as_leeftijd(cls, mks_geboortedatum):
+    def _get_age(cls, now, birthday):
+        """
+        Age calculation that takes leap years into account
+
+        :param now:
+        :param birthday:
+        :return:
+        """
+        try:
+            day = birthday.replace(year=now.year)
+        except ValueError:
+            # Fails in leap year, set day to first of next month
+            day = birthday.replace(year=now.year, month=birthday.month + 1, day=1)
+
+        if day > now:
+            return now.year - birthday.year - 1
+        else:
+            return now.year - birthday.year
+
+    @classmethod
+    def as_leeftijd(cls, mks_geboortedatum,  ind_onvolledige_datum=None, overlijdensdatum=None):
+        """
+        birthday string as age
+
+        Accepts birthday strings with unknown month. The age is unknown in the same month
+        but otherwise simple known as if the birthday was on a arbitrary day in the month
+
+        :param mks_geboortedatum:
+        :param is_overleden:
+        :return:
+        """
+        if not mks_geboortedatum or overlijdensdatum:
+            return None
+
+        incomplete_date_indicator = IncompleteDateIndicator(ind_onvolledige_datum)
+        if not (incomplete_date_indicator.is_jaar_bekend() and incomplete_date_indicator.is_maand_bekend()):
+            return None
+
         if cls._is_mks_datum(mks_geboortedatum):
             # Interpret all dates as dates in the current timezone
             now = _today()
-            birthday = datetime.datetime.strptime(mks_geboortedatum, cls._MKS_DATUM_PARSE_FORMAT)
-            return relativedelta(now, birthday).years
+            birthday = datetime.datetime.strptime(mks_geboortedatum, cls._MKS_DATUM_PARSE_FORMAT).date()
+            day_is_unknown = not incomplete_date_indicator.is_dag_bekend()
+            if not (day_is_unknown and now.month == birthday.month):
```

Comment:
Lot of negating going on here. Not really clear at first sight what's meant here.

Reply:
And so for this one